starknet_devnet/blocks.py

```diff
@@ -77,22 +77,31 @@ class DevnetBlocks:
     def __init__(self, origin: Origin, lite=False) -> None:
         self.origin = origin
         self.lite = lite
-        self.__num2block: Dict[int, StarknetBlock] = {}
+        self.__hash2block: Dict[int, StarknetBlock] = {}
         self.__state_updates: Dict[int, BlockStateUpdate] = {}
-        self.__hash2num: Dict[str, int] = {}
+        self.__num2hash: Dict[int, int] = {}
         self.__pending_block: StarknetBlock = None
         self.__pending_state_update: BlockStateUpdate = None
         self.__pending_signatures: Sequence[List[int]] = None
         self.__state_archive = MemoryStateArchive()
 
     async def get_last_block(self) -> StarknetBlock:
         """Returns the last block stored so far."""
-        number_of_blocks = self.get_number_of_blocks()
+        number_of_blocks = self.get_number_of_not_aborted_blocks()
         return await self.get_by_number(number_of_blocks - 1)
 
     def get_number_of_blocks(self) -> int:
         """Returns the number of blocks stored so far."""
-        return len(self.__num2block) + self.origin.get_number_of_blocks()
+        return len(self.__hash2block) + self.origin.get_number_of_blocks()
+
+    def get_number_of_not_aborted_blocks(self) -> int:
+        """Returns the number of not aborted blocks."""
+        not_aborted_blocks = {
+            key: val
+            for key, val in self.__hash2block.items()
+            if val.status is not BlockStatus.ABORTED
+        }
+        return len(dict(not_aborted_blocks)) + self.origin.get_number_of_blocks()
```

Comment:
Unnecessary dict wrapper, not_aborted_blocks is already a dict

Reply:
Fixed in 4581da40b77aa79ba1f6a3aabd024a63f764feca